modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -567,25 +596,47 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
       assert(txParams.rbfTxIds.length === 1);
 
       const txToBeReplaced = await wallet.getTransaction({ txHash: txParams.rbfTxIds[0], includeRbf: true });
-      expectedOutputs = txToBeReplaced.outputs
-        .filter((output) => output.wallet !== wallet.id()) // For self-sends, the walletId will be the same as the wallet's id
-        .map((output) => {
-          return { amount: BigInt(output.valueString), address: this.canonicalAddress(output.address) };
-        });
+      expectedOutputs = txToBeReplaced.outputs.flatMap(
+        (output: { valueString: string; address?: string; wallet?: string }) => {
+          // For self-sends, the walletId will be the same as the wallet's id
+          if (output.wallet === wallet.id()) {
+            return [];
+          }
+          // In the case that this is an OP_RETURN output or another non-encodable scriptPubkey, we dont have an address.
+          // We will verify that the amount is zero, and if it isnt then we will throw an error.
+          if (!output.address) {
+            if (output.valueString !== '0') {
+              throw new Error(`Only zero amounts allowed for non-encodeable scriptPubkeys: ${JSON.stringify(output)}`);
+            }
+            return [{ amount: BigInt(0) }];
+          }
+          return [{ amount: BigInt(output.valueString), address: this.canonicalAddress(output.address) }];
```

Comment:
pretty sure we can solve this with `assertValidTransactionRecipient()`/`toCanonicalTransactionRecipient()`